examples/rtl_433_mqtt_hass.py

```diff
@@ -619,10 +686,16 @@ def publish_config(mqttc, topic, model, instance, mapping):
     discovery_timeouts[path] = now + args.discovery_interval
 
     config = mapping["config"].copy()
-    config["name"] = object_name
-    config["state_topic"] = topic
-    config["unique_id"] = object_name
-    config["device"] = { "identifiers": object_id, "name": object_id, "model": model, "manufacturer": "rtl_433" }
+
+    if device_type == 'device_automation':
```

Comment:
I would like to see a comment explaining the split.   I don't yet understand the details of MQTT discovery, but I find it hard to convince myself that this is right.  (That's not an assertion it isn't.)

Reply:
device_automation has a different configuration structure than many other types. Would a link to https://www.home-assistant.io/integrations/device_trigger.mqtt/ suffice?

Reply:
I've added a comment for this.